### PR TITLE
Support server paths

### PR DESCRIPTION
**Pull request by Gaztin**
For context, see #1129 

TLDR: Server paths (i.e. "//server/files") were treated like any non-normalized path with duplicate slashes.

### CODE REVIEW

File: src/host/path_getrelative.c [hunk 1]
```diff
@@ -67,6 +67,14 @@ int path_getrelative(lua_State* L)
 		return 1;
 	}
 
+	/* Relative paths within a server can't climb outside the server root.
+	* If the paths don't share server name, return the absolute path. */
+	if (src[0] == '/' && src[1] == '/' && last == 1) {
```
[the remark this answers is not in the document]
Comment (Gaztin) [reply src/host/path_getrelative.c]: But isn't `src[0] != '\0' && src[0] == '/'` redundant? if `src[0] == '/'` then it's guaranteed to not be 0

[the remark this answers is not in the document]
Comment (Gaztin) [reply src/host/path_getrelative.c]: Additionally, empty strings are covered in the previous check.

[the remark this answers is not in the document]
Comment (samsinsane) [reply src/host/path_getrelative.c]: Haha, not really sure what I was thinking when I made that comment. Sorry about that!

File: src/host/path_normalize.c [hunk 2]
```diff
@@ -70,8 +70,8 @@ static void* normalize_substring(const char* str, const char* endPtr, char* writ
 			}
 		}
 
-		/* add to the result, filtering out duplicate slashes */
-		if (ch != '/' || last != '/') {
+		/* add to the result, filtering out duplicate slashes, except when they are leading slashes */
+		if (str == &source[1] || (ch != '/' || last != '/')) {
```
Comment (samsinsane) [inline src/host/path_normalize.c]: I think this is the line I was meant to comment on. Is `source` guaranteed to not be an empty string here?

Comment (Gaztin) [reply src/host/path_normalize.c]: Without testing, I believe so. if the string is empty, then `str` will be equal to `endPtr` and the loop condition `while (str != endPtr)` will fail.

Comment (Gaztin) [reply src/host/path_normalize.c]: After some testing I found out that it never attempt to normalize an empty string, thanks to https://github.com/premake/premake-core/blob/a713eccf96e352568b1798c71d14d4c5221a4a73/src/host/path_normalize.c#L108

**Pull request merged**